File: ex/ch04/ExTutotest/app/src/main/java/com/ehappy/extutotest/MainActivity.java
package com.ehappy.extutotest;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Button btn1;
    private TextView txt1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn1=(Button)findViewById(R.id.btnButton);
        txt1= (TextView)findViewById(R.id.textView);
        btn1.setOnClickListener(listener);
     }

    private Button.OnClickListener listener=new Button.OnClickListener() {

        @Override
        public void onClick(View view) {
            
        }
    };
}

File: ex/AppendixB/ExMileToKm/app/src/main/java/com/ehappy/exmiletokm/MainActivity.kt
package com.ehappy.exmiletokm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 設定 btnTran 元件 onClick 事件為  btnTranListener
        btnTran.setOnClickListener(btnTranListener)
    }

    // 定義  onClick() 事件的方法
    val btnTranListener = View.OnClickListener { view ->
        val miles=edtMile.text.toString().toDouble()
        var km:Double = 1.61*miles
        txtKm.text="時速=  $km 公里"
    }
}